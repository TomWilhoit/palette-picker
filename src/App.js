import React, { Component } from 'react';
import { Route, Switch } from "react-router-dom";
import { connect } from 'react-redux';
import { addProjects } from './Actions';
import { addPalettes } from './Actions'
import { fetchData } from './Utils/API'
import PropTypes from "prop-types"
import './App.scss';

export class App extends Component {
  
  componentDidMount( ){
    this.fetchProjectsData();
    this.fetchPalettesData()
  }

  fetchProjectsData = async () => {
    const url = "http://localhost:3000/api/v1/projects"
    const response = await fetchData(url)
    await this.props.addProjects(response)
  }

  fetchPalettesData = async () => {
    const url = "http://localhost:3000/api/v1/palettes"
    const result = await fetchData(url)
    await this.props.addPalettes(result)
  }

  testCI = () => {
    console.log('this tests the TravisCI changes')
  }

  render() {
      return (
        <div className="App">
          <header>Tom and Mason are Really Good at Programming</header>
        </div>
      );
    }
  }

  App.propTypes = {
    locations: PropTypes.array,
    palettes: PropTypes.array
  };

export const mapStateToProps = state => ({
  projects: state.projects,
  palettes: state.palettes
});

export const mapDispatchToProps = dispatch => ({
  addProjects: projects => dispatch(addProjects(projects)),
  addPalettes: palettes => dispatch(addPalettes(palettes))
})


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);

