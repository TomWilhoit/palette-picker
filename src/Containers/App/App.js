import React, { Component } from "react";
import { connect } from "react-redux";
import { addProjects } from "../../Actions";
import { addPalettes } from "../../Actions";
import { fetchData } from "../../Utils/API";
import PropTypes from "prop-types";
import { Loading } from "../../Components/Loading/Loading";
import { Header } from "../../Components/Header/Header";
import PalettePicker from "../PalettePicker/PalettePicker";

export class App extends Component {
  constructor() {
    super();
    this.state = {
      error: "",
      isLoading: false
    };
  }

  componentDidMount = async () => {
    this.toggleLoading();
    console.log(process)
    // console.log(process.env.REACT_APP_BACKEND_URL)
    try {
      const projects = await fetchData("https://palette-api-tm.herokuapp.com/" + "api/v1/projects");
      const palettes = await fetchData("https://palette-api-tm.herokuapp.com/" + "api/v1/palettes");
      await this.storeData(projects, palettes);
    } catch (error) {
      this.setState({ error: error.message });
    }
    this.toggleLoading();
  }

  storeData = (projects, palettes) => {
    this.updateProjects(projects);
    this.updatePalettes(palettes);
  }

  updateProjects = projects => {
    this.props.addProjects(projects);
  };

  updatePalettes = palettes => {
    this.props.addPalettes(palettes);
  };

  toggleLoading = () => {
    this.setState({ isLoading: !this.state.isLoading });
  };

  render() {
    return (
      <div className="App">
        <div className="head">
          <Header />
          {this.state.error && this.state.error}
          {this.state.isLoading && <Loading />}
        </div>
        <div className="main">
          <PalettePicker />
        </div>
      </div>
    );
  }
}

App.propTypes = {
  locations: PropTypes.array,
  palettes: PropTypes.array
};

export const mapStateToProps = state => ({
  projects: state.projects,
  palettes: state.palettes
});

export const mapDispatchToProps = dispatch => ({
  addProjects: projects => dispatch(addProjects(projects)),
  addPalettes: palettes => dispatch(addPalettes(palettes))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);