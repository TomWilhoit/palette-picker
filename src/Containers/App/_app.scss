* {
  box-sizing: border-box;
  margin: 0;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  overflow: hidden;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.App {
  background-color: $background;
  color: #263674;
  display: grid;
  font-family: 'Josefin Sans', sans-serif;
  grid-template-rows: 10% 90%;
  grid-template-areas: 
    "head"
    "main";
  height: 100vh;
  text-align: center;
}

.head {
  grid-area: head;
  display: grid;
  grid-template-columns: 18% 64% 18%;
  grid-template-areas: 
    "left center right";
}

.left-head {
  grid-area: left;
}

.right-head {
  display: flex;
  flex-direction: column;
  grid-area: right;
  justify-content: space-around;
}

.center-head {
  display: flex;
  flex-direction: column;
  grid-area: center;
  justify-content: space-around;
}


.main {
  grid-area: main;
}