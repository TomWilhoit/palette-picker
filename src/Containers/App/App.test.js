import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

describe('App', () => {
  describe('app load', () => {
    it('should match snapshot', () => {
  
    })
    it('should have default state', () => {
  
    })
  })
  describe('mapStateToProps', () => {

  })
  describe('mapDispatchToProps', () => {
    
  })
})


// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });
