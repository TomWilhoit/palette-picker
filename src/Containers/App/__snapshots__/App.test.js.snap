// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App on load should match snapshot 1`] = `
<div
  className="app"
>
  <div
    className="head"
  >
    <div
      className="left-head"
    />
    <div
      className="center-head"
    >
      <Header />
      <div
        className="info-btn"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-info"
        />
      </div>
    </div>
    <div
      className="right-head"
    >
      <Loading />
    </div>
  </div>
  <div
    className="main"
  >
    <ConnectFunction
      clearError={[Function]}
      setError={[Function]}
    />
  </div>
  <div
    className="info-pop-up closable"
  >
    <Info
      hideInfo={[Function]}
    />
  </div>
  <div
    className="modal-overlay"
    onClick={[Function]}
  />
</div>
`;
